### ISSUE TITLE
Role inheritance with pattern matching domains

### ISSUE THREAD

**Issue by lfarrel6** [casbin/casbin#473]
### Background Context

I need to configure a RBAC control system with domains. Users can be members of many domains and have a specific role within each domain. I want to configure the roles in the system to have a role hierarchy within the scope of these domains. 

For example a user with role guest in domain/1 would have read only access. A user with role admin in the same domain should inherit the guest privileges while also having create, update and delete abilities.

The roles will be consistent throughout the domains but the domains will be dynamically generated. Because of this, the policy definitions have to be parameterised (i.e. `p, guest, domain/:domainId, resource, read`)

I can produce the desired behaviour by assigning the hierarchy of roles to users explicitly:
`g, bob, guest, domain/1`, `g, bob, admin, domain/1` 
But ideally I would be able to assigning the hierarchy to the roles directly using `g2, admin, guest`

### Repro configuration

**model config**
```ini
[request_definition]
r = sub, dom, obj, act 

[policy_definition]
p = sub, dom, obj, act

[role_definition]
g = _, _, _ 
g2 = _, _   

[policy_effect]
e = some(where (p.eft == allow)) && !some(where (p.eft == deny))

[matchers]
m = g(r.sub, p.sub, r.dom) && r.obj == p.obj && r.act == p.act
```

**policies**
```csv
p, guest, domain/:domainId, resource, read
p, admin, domain/:domainId, resource, delete

g2, admin, guest
g, alice, guest, domain/1
g, bob, admin, domain/1
```

**requests**
```
1. alice, domain/1, resource, read => true
2. bob, domain/1, resource, read => false
3. alice, domain/1, resource, delete => false
4. bob, domain/1, resource, delete => true
```

In the requests above, request 2 should resolve as true. 

How can I configure my matcher to account for the inheritance of `guest` by `admin`?

**Comment by hsluoyz** [comment]
@lfarrel6 I think you want this feature: https://github.com/casbin/casbin/pull/371

Policy should change to:

```csv
p, guest, domain/:domainId, resource, read
p, admin, domain/:domainId, resource, delete

g, admin, guest, *
g, alice, guest, domain/1
g, bob, admin, domain/1
```

**Comment by lfarrel6** [comment]
Ok. I am I correct in saying that #371 is not yet supported?

**Comment by hsluoyz** [comment]
Well, the temp conclusion for that feature is: you can use: https://github.com/dovics/domain-role-manager to enable that feature. If you are in a hurry, you can use it right now.

We are also working on merging it back to the core library (this repo) these days, see: https://github.com/casbin/casbin/pull/371#issuecomment-635948706 but it requires some time.

**Comment by lfarrel6** [comment]
Cool, thanks for your help!

**Comment by WesleyBatista** [comment]
Hey there :wave:, did the decision of including this feature to the core library change in the meantime? the mentioned repository  https://github.com/dovics/domain-role-manager is not available anymore.

**Comment by WesleyBatista** [comment]
Oh, I see now why you said _"but it requires some time"_ 

The code is merged on master: https://github.com/casbin/casbin/blob/6dcd52b3eecf8fc27ce065c7b3fda300f051f468/util/builtin_operators.go#L213-L221

but not yet available on [v2.19.7](https://github.com/casbin/casbin/blob/v2.19.7/util/builtin_operators.go).

sorry :facepalm:

**Comment by hsluoyz** [comment]
@WesleyBatista can you send a new GitHub issue for your question?
